### DISCUSSION TITLE
请问一下5.X版本升级后是不支持针对维度设置size通道了吗

### DISCUSSION THREAD

**Discussion by zhouhe-1476452306** [antvis/G2#5158] [Q&A]
4.X 设置颜色通道根据不同year 散点的大小有改变
![image](https://github.com/antvis/G2/assets/75061709/8daeab2d-870d-4a5f-8201-fe2e34b3b437)
5.0.11  设置颜色通道为维度不是指标时，legendContinuous.js 该文件会有报错，再加上color通道也只能区分颜色大小没有改变
![image](https://github.com/antvis/G2/assets/75061709/912ecad6-c842-4e57-a1c3-cf020cf8a59f)

**Comment by hustcc** [comment]
5.0 中 size 仅仅支持数值类型，也就是连续的数据。可以这么做：

```ts
.encode('size', (d) => Number(d.year))
```

**Reply by pearmini** [reply]
这个确实现在不支持，但应该需要支持一下。

**Reply by pearmini** [reply]
可以跟进这个 issue：https://github.com/antvis/G2/issues/5166

**Comment by pearmini** [comment]
> 5.0.11 设置颜色通道为维度不是指标时，legendContinuous.js 该文件会有报错 

这个有具体的 demo 吗？

**Reply by zhouhe-1476452306** [reply]
5.0.11版本的demo地址 **https://g2.antv.antgroup.com/examples/general/point/#point-basic** 就是基础散点图的示例，根据‘gender’字段设置颜色通道
![image](https://github.com/antvis/G2/assets/75061709/59959877-1a28-4472-86fd-e06cfc8cdc12)

**Reply by pearmini** [reply]
嗯嗯，应该是不设置 color 通道，只设置 size 通道的时候会报错。

**Reply by zhouhe-1476452306** [reply]
对的 当时是这样的情况下有问题

**Reply by pearmini** [reply]
可以跟进这个 issue：https://github.com/antvis/G2/issues/5169
